gems/sorbet-runtime/test/types/virally_install_hooks.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+require_relative '../test_helper'
+
+class Opus::Types::Test::VirallyInstallHooksTest < Critic::Unit::UnitTest
+  it "does not give the hooks when including a non-viral module" do
+    m = Module.new do
+      extend T::Sig
+      sig {void}
+      def foo; end
+    end
+    assert_equal(true, T::Private::Methods.instance_variable_get(:@installed_hooks).include?(m))
+    c = Class.new do
+      include m
+    end
+    assert_equal(false, T::Private::Methods.instance_variable_get(:@installed_hooks).include?(c))
+  end
+
+  it "does not give the hooks when extending a non-viral module" do
+    m = Module.new do
+      extend T::Sig
+      sig {void}
+      def foo; end
+    end
+    assert_equal(true, T::Private::Methods.instance_variable_get(:@installed_hooks).include?(m))
+    c = Class.new do
+      extend m
+    end
+    assert_equal(false, T::Private::Methods.instance_variable_get(:@installed_hooks).include?(c))
+  end
+
+  it "gives the hooks when including a viral module" do
+    m = Module.new do
+      T::Private::Methods.instance_variable_get(:@virally_install_hooks) << self
+      extend T::Sig
+      sig {void}
+      def foo; end
+    end
+    assert_equal(true, T::Private::Methods.instance_variable_get(:@installed_hooks).include?(m))
+    c = Class.new do
+      include m
+    end
+    assert_equal(true, T::Private::Methods.instance_variable_get(:@installed_hooks).include?(c))
+  end
+
+  it "gives the hooks when extending a viral module" do
+    m = Module.new do
+      T::Private::Methods.instance_variable_get(:@virally_install_hooks) << self
+      extend T::Sig
+      sig {void}
+      def foo; end
+    end
+    assert_equal(true, T::Private::Methods.instance_variable_get(:@installed_hooks).include?(m))
+    c = Class.new do
+      extend m
+    end
```

Comment:
If one of these Classes has

```
def included; end
def extended; end
```

i.e., does not call `super`, does that prevent the things you introduced in this branch from working?

Reply:
Almost certainly. Should I test for this explicitly? Should we try to prevent this?

Reply:
It would seem that we definitely want to prevent any way to change a final method.